layers/+lang/go/config.el

```diff
@@ -36,3 +36,6 @@
 
 (defvar go-use-test-args ""
   "Additional arguments to be supplied to `go test` during runtime.")
+
+(defvar go-lsp-prefer-flymake :none
```

Comment:
IMO variable should be called go-use-lsp-ui-flychecker or something like that and should be boolean. I think you should preserve the current default behaviour - imagine that you are go layer user and you have used lsp-ui for flychecker and now after update spacemacs and you will end up with no flychecker enabled. Also, it will be consistent with the default behaviour of the other layers that use language server. 

You have to document the variable in the README.org.

Reply:
@yyoncho I definitely appreciate your point. Although, in my testing it didn't break other common Spacemacs Go configs. It also fixed an issue where the layer wasn't fully configuring things properly based on what it was doing that.

Considering that, maybe instead I'll remove the configuration completely, and if you are using `lsp` and `golangci-lint` it uses `:none`. Otherwise it does nothing. The reason I think this is appropriate is that currently turning on `golangci-lint` when in `lsp` mode will silently fail.

Reply:
Make sense, this will cover my concerns.

Reply:
👍